### DISCUSSION TITLE
Module to configure a layout for a product

### DISCUSSION THREAD

**Discussion by IGNF-Xavier** [geonetwork/geonetwork-ui#929] [Ideas]
I suggest considering a module allowing you to configure a template from any type of metadata as long as the schema is provided.
Moreover, these schema are often similar whether they are reuse, services, organizations or any other type of objects to be referenced.

For the community, can we have an estimation of the cost of this feature (if recognized as useful by the community)

**Comment by jahow** [comment]
The first release of the editor (https://github.com/geonetwork/geonetwork-ui/issues/923) will theoretically include this, although the concept of "template" is something that we have tried to avoid as long as possible. Templates were ubiquitous in the previous GeoNetwork editor because they would offer a starting point for the metadata author, but the new editor might offer enough simplicity and abstraction to not require this altogether.

**Comment by IGNF-Xavier** [comment]
My suggestions are for the rendering in the catalog not for the metadata editor (even if it would be useful for certain use-case)

**Reply by jahow** [reply]
Then could you please clarify what you mean by "template"? Is it a page layout? And what does "schema" mean in that context? I assume it's not a "metadata schema" as in ISO19139 etc.

**Reply by IGNF-Xavier** [reply]
i probably don't use the correct words.
For example, we will be in ISO19139 to render this service : 
![image](https://github.com/geonetwork/geonetwork-ui/assets/98382923/2afdfebd-4829-498a-9786-52bfa96cb378)
But it is possible that we won't expose all the attributes of the metadata or different attributes because which tha applies to a data won't necessary apply to a reuse or a software

**Reply by jahow** [reply]
Yes so having different layouts with different information is what would really make sense if we want to natively show services in the datahub. Data preview wouldn't make sense for a service; on the other hand, instructions and tools to use it would be much more relevant.

I would argue this is still very much a functional/design topic, as the GN-UI community has to agree on what services (and apps etc.) would look like in the Datahub before even giving a rough estimation.

From what I have seen and heard, there is definitely a strong interest in the community.
